lxc/config.go

```
@@ -141,7 +191,341 @@ func (c *configCmd) run(config *lxd.Config, args []string) error {
+func doProfileSet(client *lxd.Client, p string, args []string) error {
+	// we shifted @args so so it should read "<key> [<value>]"
+	if len(args) < 1 {
+		return fmt.Errorf(gettext.Gettext("Bad config command"))
+	}
+
+	key := args[0]
+	var value string
+	if len(args) < 2 {
+		value = ""
+	} else {
+		value = args[1]
+	}
+	err := client.SetProfileConfig(p, key, value)
+	return err;
+}
+
+func doProfileCopy(client *lxd.Client, p string, args []string) error {
+	if len(args) < 4 {
+		return fmt.Errorf(gettext.Gettext("Bad config command"))
+	}
+	return fmt.Errorf("profile copy not yet implemented")
+}
+
+func doProfileList(config *lxd.Config, args []string) error {
+	var remote string
+	if len(args) > 2 {
+		var name string
+		remote, name = config.ParseRemoteAndContainer(args[2])
+		if name != "" {
+			return fmt.Errorf("Cannot provide container name to list")
+		}
+	} else {
+		remote = config.DefaultRemote
+	}
+
+	client, err := lxd.NewClient(config, remote)
+	if err != nil {
+		return err
+	}
+
+	/*
+	var filters []string
+	if len(args) > 3 {
+		filters = []string{}
+	} else {
+		filters = args[3:]
+	}
+	*/
+
+	profiles, err := client.ListProfiles()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%s\n", strings.Join(profiles, "\n"))
+	return nil
+}
+
+func deviceAdd(config *lxd.Config, which string, args []string) error {
+	if len(args) < 5 {
+		return fmt.Errorf(gettext.Gettext("Bad config command"))
+	}
+	remote, name := config.ParseRemoteAndContainer(args[2])
+
+	client, err := lxd.NewClient(config, remote)
+	if err != nil {
+		return err
+	}
+
+	devname := args[3]
+	devtype := args[4]
+	var props []string
+	if len(args) > 5 {
+		props = args[5:]
+	} else {
+		props = []string{}
+	}
+
+	var resp *lxd.Response
+	if which == "profile" {
+		resp, err = client.ProfileDeviceAdd(name, devname, devtype, props)
+	} else {
+		resp, err = client.ContainerDeviceAdd(name, devname, devtype, props)
+	}
+	if err != nil {
+		fmt.Printf(gettext.Gettext("Error: %s\n"), err)
```

I think we can just return the error here and it'll get printed out for us.